src/core/core.scale.js

```diff
@@ -333,7 +333,7 @@ module.exports = Element.extend({
 		var me = this;
 		var context = me.ctx;
 		var tickOpts = me.options.ticks;
-		var labels = labelsFromTicks(me._ticks);
+		var labels = tickOpts.autoSkip ? me._autoSkip(me.getTicks()) : me.getTicks();
```

Comment:
The `labels` variable looks like it is now holding ticks. Why do you think you can remove `labelsFromTicks` ?

Also, the tests are failing